rllib/evaluation/sample_batch_builder.py

```diff
@@ -196,13 +196,14 @@ def postprocess_batch_so_far(
                 raise ValueError(
                     "Batches sent to postprocessing must only contain steps "
                     "from a single trajectory.", pre_batch)
-            post_batches[agent_id] = policy.postprocess_trajectory(
-                pre_batch, other_batches, episode)
             # Call the Policy's Exploration's postprocess method.
+            post_batches[agent_id] = pre_batch
             if getattr(policy, "exploration", None) is not None:
                 policy.exploration.postprocess_trajectory(
                     policy, post_batches[agent_id],
                     getattr(policy, "_sess", None))
+            post_batches[agent_id] = policy.postprocess_trajectory(
```

Comment:
Why this change?

Reply:
Good question. Curiosity has to change the rewards before e.g. PPO calculates the advantages from those (otherwise, the intrinsic rewards would have no effect on the loss).
`Exploration.postprocess_trajectory()` is only used so far inside the ParameterNoise exploration class, where it doesn't matter, whether we call it before or after `Policy.postprocess_trajectory()`.
Maybe we should have two methods: before and after.